fix caffe model convert leakyrelu

vsooda opened the pull request:
I tried to convert an trained caffe model to mxnet, but the predict result is wrong. Finally I found what cause the error.

I had some activation define in prototxt such as:

```
layer {
    bottom: "layer23-conv"
    top: "layer23-conv"
    name: "layer23-act"
    type: "ReLU"
    relu_param {
        negative_slope: 0.1
    }
}
```

in fact, according to caffe document:  

> Given an input value x, The ReLU layer computes the output as x if x > 0 and negative_slope * x if x <= 0.

so this is `LeakyRelu`, not `relu`. but the convert tools treat this as relu, so error happen.

I fixed this error, and the result is now the same with caffe.

piiswrong commented:
is negative_slope always present? In which version was this introduced? Is it better to check hasattr(param, 'negative_slope') first?

vsooda commented:
@piiswrong I check caffe commit log, negative_slope was introduced in 7.22.2014. this is a long time bug...

although in current version, the negative_slope is always present, but it can also be removed in the future, so add 'hasattr' judgement is more safe. I had updated the pull request

piiswrong commented:
Have you tested this?
param is used before assignment.

Please verify this on a actual protobuf since it's not covered by test cases

vsooda commented:
@piiswrong  I am embarrassed for this mistake. I fixed it on another old repo, then copied the fix code to this branch to make this pr. one line was missed along this way.

I had fixed this error, and tested the converted model, it is ok now.

piiswrong commented:
Cool. Thanks
